### PR TITLE
feature: INSERT OR UPDATE (upsert) support

### LINKED ISSUE #202: Support QueryBuilder::upsert()
Use the recently introduced `INSERT OR UPDATE` https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/releases/tag/v1.5.15.

### PR DESCRIPTION
**Pull request by taka-oyama**
Closes #202

### CODE REVIEW

File: src/Query/Builder.php [hunk 1]
```diff
@@ -55,6 +56,33 @@ public function updateOrInsert(array $attributes, array $values = [])
         return (bool) $this->take(1)->update(Arr::except($values, array_keys($attributes)));
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function upsert(array $values, $uniqueBy = null, $update = null)
```
Comment (tyabu12) [inline src/Query/Builder.php:62]: `$uniqueBy` of `parant:upsert()` is `array|string`. It does not seem to be null.

https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Query/Builder.php#L3674

**Review by tyabu12 — CHANGES_REQUESTED** [review]

File: src/Query/Builder.php (hunk 1 above)
Comment (taka-oyama) [reply src/Query/Builder.php:62]:
```suggestion
    public function upsert(array $values, $uniqueBy = [], $update = null)
```

**Review by tyabu12 — APPROVED** [review]
LGTM

**Comment by taka-oyama** [comment]
Thank you for the review

**Pull request merged**